ggml/src/ggml-rpc/ggml-rpc.cpp

```diff
@@ -386,6 +387,7 @@ static bool parse_endpoint(const std::string & endpoint, std::string & host, int
 // RPC request : | rpc_cmd (1 byte) | request_size (8 bytes) | request_data (request_size bytes) |
 // No response
 static bool send_rpc_cmd(const std::shared_ptr<socket_t> & sock, enum rpc_cmd cmd, const void * input, size_t input_size) {
+    std::lock_guard<std::mutex> lock(sock->send_rpc_cmd_mutex);
```

Comment:
you should also acquire the lock in the overloaded `send_rpc_cmd` function below to prevent multiple threads reading from the socket at the same time;

you may also need to use `std::recursive_mutex` for this to work